docs/src/intro.md

```diff
@@ -9,3 +9,36 @@ you want to edit this book, `mdbook serve --open` runs a local server, opens a
 formatted version of this book in your browser, and automatically refreshes the
 browser when you save a markdown file after editing.
 
+
+## Major components
+
+yk is a meta-tracing system that turns existing C interpreters into
+JIT-compiling VMs. It is comprised of two major components:
+
+  1. `ykllvm` is a fork of [LLVM](https://llvm.org/]) that must be used to:
+     compile C interpreters with the necessary extra information for yk; link
+     in the `yk` Rust library.
+
+  2. The `yk` Rust library is the core of the run-time JIT system.
+
+
+## Terminology
+
+yk utilises two different *Intermediate Representations* (IRs):
+
+  * *AOT IR* is the Ahead-Of-Time IR generated by ykllvm and embedded in the
+    binary of a C interpreter. AOT IR is similar to LLVM IR, though customised
+    and simplified for yk.
+
+  * *JIT IR* is the IR generated (from AOT IR) at run-time by yk and which is
```

Comment:
"(from an AOT IR trace)"?

Reply:
Missed this?

Reply:
I think this one is OK as-is. It makes sense both ways.

Reply:
It's tricky as at this point we haven't defined "AOT trace". I'm neutral.